### ISSUE TITLE
Rule to detect the absence of a new line between the method signature and method body

### ISSUE THREAD

**Issue by pavinduLakshan** [checkstyle/checkstyle#16989]
**Is your feature request related to a problem? Please describe.**

Is there any way to configure checkstyle to detect the absence of a new line between the method signature and method body?

**Describe the solution you'd like**

With such a rule, the following would be the behaviour.

Valid:

```java
public void testMethod() {

  // body goes here
}
```

Invalid:

```java
public void testMethod() {
  // body goes here
}
```

**Comment by romani** [comment]
New Check needs to be created.
https://checkstyle.org/writingchecks.html

If you would like to design it, here is guide on how to do this https://checkstyle.org/report_issue.html#How_to_request_new_Check.2FModule.3F

This will be very controversial, without any special logic to skip violations on simple methods.
Usually people want empty line, when method signature is very verbose and first statement is also verbose (a lot of characters on lines).
It would be awesome if you investigate this vague definition of verbose(a lot of characters) lines.

If you want to make it simple as you described, sure, but I would recommend to make it in your own repository, or at our sandbox project https://github.com/sevntu-checkstyle/sevntu.checkstyle/tree/master/sevntu-checks , where we keep controversial stuff.

**Comment by pavinduLakshan** [comment]
> New Check needs to be created. [checkstyle.org/writingchecks.html](https://checkstyle.org/writingchecks.html)
> 
> If you would like to design it, here is guide on how to do this [checkstyle.org/report_issue.html#How_to_request_new_Check.2FModule.3F](https://checkstyle.org/report_issue.html#How_to_request_new_Check.2FModule.3F)
> 
> This will be very controversial, without any special logic to skip violations on simple methods. Usually people want empty line, when method signature is very verbose and first statement is also verbose (a lot of characters on lines). It would be awesome if you investigate this vague definition of verbose(a lot of characters) lines.
> 
> If you want to make it simple as you described, sure, but I would recommend to make it in your own repository, or at our sandbox project [sevntu-checkstyle/sevntu.checkstyle@`master`/sevntu-checks](https://github.com/sevntu-checkstyle/sevntu.checkstyle/tree/master/sevntu-checks?rgh-link-date=2025-05-04T01%3A22%3A29.000Z) , where we keep controversial Checks.

Thanks for the reply. I was able to create a custom rule.

https://github.com/pavinduLakshan/checkstyle-method-body-top-padding

**Comment by romani** [comment]
nice!

we encourage people to create their own public collections of Checks, not need to push all to standard set of Checks.
People should use few jars in their build to get behavior that they need.

we can add your repo to https://checkstyle.sourceforge.io/#Additional_Checks
so people will find you more easily.
you can send PR to us to add your project to table, all site sources are in our github repository.

**Comment by romani** [comment]
we found this https://github.com/checkstyle/checkstyle/pull/16489#discussion_r2097764721
as good rule on when space is very good.

```
    private static DetailAST findRedundantNullCheck(DetailAST logicalAndNode,
                                                    DetailAST instanceOfNode) {
        DetailAST currentChild = logicalAndNode.getFirstChild();
```

better form:
```
    private static DetailAST findRedundantNullCheck(DetailAST logicalAndNode,
                                                    DetailAST instanceOfNode) {
        DetailAST currentChild = logicalAndNode.getFirstChild();
```

or 
```
    private static DetailAST findRedundantNullCheck(DetailAST logicalAndNode,
                                                    DetailAST instanceOfNode) {

        DetailAST currentChild = logicalAndNode.getFirstChild();
```
